IndoorMapsFrontend/src/components/pages/Directory.tsx

```diff
@@ -0,0 +1,63 @@
+import { Button, Group, JsonInput } from "@mantine/core";
+import React, { useEffect } from "react";
+import { graphql, usePreloadedQuery, useQueryLoader } from "react-relay";
+import type { PreloadedQuery } from 'react-relay';
+import type { DirectoryGetAllBuildingsQuery } from "./__generated__/DirectoryGetAllBuildingsQuery.graphql";
+
+type Props = {
+
+}
+const GetAllBuildings = graphql`
+    query DirectoryGetAllBuildingsQuery {
+    allBuildings {
+      id
+      title
+      description
+    }
+  }
+`
+
+const Directory = ({ }: Props) => {
+    const [
+        queryReference,
+        loadQuery,
+    ] = useQueryLoader<DirectoryGetAllBuildingsQuery>(
+        GetAllBuildings,
+    );
+
+    useEffect(() => {
+        loadQuery({});
+    }, []);
+
+    return (
+        <>
+            <React.Suspense fallback="Loading">
+                {queryReference != null
+                    ? <NameDisplay queryReference={queryReference} />
+                    : null
+                }
+            </React.Suspense>
+        </>
+    )
+}
+
+type NameDisplayProps = {
+    queryReference: PreloadedQuery<DirectoryGetAllBuildingsQuery>
+}
+
+function NameDisplay({ queryReference }: NameDisplayProps) {
+    const data = usePreloadedQuery(GetAllBuildings, queryReference);
+
+    const buildingListElements = data.allBuildings.map((building, i) => {
```

Comment:
Nit: `data.allBuildings.map({title, id, description} => {`

De-structuring right at top to keep code clean

Reply:
Incorporated into newest branch:
https://github.com/Theo-s-Meta-Capstone/IndoorMaps/commit/35a1a21e0b09010fa71afbf09afce84627307f1f